vite.config.ts

```diff
@@ -0,0 +1,84 @@
+import {defineConfig} from 'vite';
+import react from '@vitejs/plugin-react';
+import {resolve} from 'path';
+
+const commitHash = process.env.VITE_COMMIT_HASH;
+function linkedDependencyHMR() {
+  return {
+    name: 'linked-dependency-hmr',
+    handleHotUpdate({file, server}) {
+      if (file.includes('node_modules/vega-lite/src/') || file.includes('node_modules/vega-lite/build/')) {
+        const module = server.moduleGraph.getModuleById('vega-lite');
+        if (module) {
+          server.reloadModule(module);
+        }
+        server.ws.send({
+          type: 'full-reload',
```

Comment:
This isn't doing full HMR, right? We just happen to have the state in local storage.